### PR TITLE
Short fiber example: fiber orientation tensor

### PR DESCRIPTION
**Pull request by FedericoNegri**

### CODE REVIEW

File: examples/9_short_fiber_orientation_tensor.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
+"""
+.. _short_fiber_orientation_tensor:
+
+Plot of the orientation tensor for short fiber composites
+---------------------------------------------------------
+
+This example shows how reconstruct and plot the components
```
Comment (roosre) [inline examples/9_short_fiber_orientation_tensor.py:7]:
```suggestion
This example shows how to reconstruct and plot the components
```

File: examples/9_short_fiber_orientation_tensor.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    ).as_matrix()
+    d1 = a11.get_entity_data_by_id(eid)[0]
+    d2 = a22.get_entity_data_by_id(eid)[0]
+    D = np.diag([d1, d2, max(1.0 - d1 - d2, 0.0)])
+    aRot = np.matmul(np.matmul(V, D), V.transpose())
+    fiber_orientation_tensor.append(
+        [aRot[0, 0], aRot[1, 1], aRot[2, 2], aRot[0, 1], aRot[1, 2], aRot[0, 2]], eid
+    )
+
+fiber_orientation_tensor_fc = dpf.operators.utility.field_to_fc(
```
Comment (roosre) [inline examples/9_short_fiber_orientation_tensor.py:102]: Out of curiosity, does field_to_fc stand for field to field container?

**Review by roosre — APPROVED** [review]

File: examples/9_short_fiber_orientation_tensor.py (hunk 2 above)
Comment (FedericoNegri) [reply examples/9_short_fiber_orientation_tensor.py:102]: Exactly

**Comment by FedericoNegri** [comment]
@janvonrickenbach @greschd I downloaded the html doc artifact and was surprised to see that the Axx and Ayy plot have inverted titles! 
image
I've already noticed in the past the order of saved images to be different than the actual commands, and I'm fine with that. But this is an issue. Ever seen it before?

These are the expected plots
image

**Comment by janvonrickenbach** [comment]
> @janvonrickenbach @greschd I downloaded the html doc artifact and was surprised to see that the Axx and Ayy plot have inverted titles! image I've already noticed in the past the order of saved images to be different than the actual commands, and I'm fine with that. But this is an issue. Ever seen it before?
> 
> These are the expected plots image

@PProfizi: We get strange results when plotting results on the same mesh. To me this looks related to https://github.com/pyansys/pydpf-core/issues/632.

File: examples/9_short_fiber_orientation_tensor.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+# Set up analysis
+# ~~~~~~~~~~~~~~~
+# Setting up the analysis consists of loading Ansys libraries, connecting to the
+# DPF server, and retrieving the example files.
+#
+# Load Ansys libraries.
+
+import ansys.dpf.core as dpf
+import numpy as np
+from scipy.spatial.transform import Rotation
```
Comment (janvonrickenbach) [inline examples/9_short_fiber_orientation_tensor.py:24]: Note that scipy is installed only for the doc build. A user that just installs pydpf-composites does not get scipy and will have to install scipy. I think it is good that scipy is not a dependency of pydpf-composites, but it probably makes sense to add a comment here.

**Review by janvonrickenbach — APPROVED** [review]

**Comment by PProfizi** [comment]
> > @janvonrickenbach @greschd I downloaded the html doc artifact and was surprised to see that the Axx and Ayy plot have inverted titles! image I've already noticed in the past the order of saved images to be different than the actual commands, and I'm fine with that. But this is an issue. Ever seen it before?
> > These are the expected plots image
> 
> @PProfizi: We get strange results when plotting results on the same mesh. To me this looks related to [pyansys/pydpf-core#632](https://github.com/pyansys/pydpf-core/issues/632).

Hi @janvonrickenbach, looking at the scalar bars between your "expected" and the "current", not only are the titles inverted but the maximum and minimum are not the same. I will take a look but I need to know how you got your "expected" reference. Is that with DPF-Core 0.7.4?

**Comment by FedericoNegri** [comment]
@PProfizi Reference results were obtained by running the notebook locally step-wise in a venv with DPF-Core 0.7.4.

**Comment by greschd** [comment]
Another way I managed to produce the wrong results is by creating three different meshes to plot on (while keeping the rest the same):

```patch
@@ -50,6 +50,12 @@ data_sources.set_result_file_path(composite_files_on_server.rst)
 model = dpf.Model(data_sources=data_sources)
 mesh = model.metadata.meshed_region
 
+model2 = dpf.Model(data_sources=data_sources)
+mesh2 = model2.metadata.meshed_region
+
+model3 = dpf.Model(data_sources=data_sources)
+mesh3 = model3.metadata.meshed_region
+
 # %%
 # Plot input data
 # ~~~~~~~~~~~~~~~
@@ -112,6 +118,16 @@ aXX = fiber_orientation_tensor_fc.select_component(0)[0]
 aYY = fiber_orientation_tensor_fc.select_component(1)[0]
 aXZ = fiber_orientation_tensor_fc.select_component(5)[0]
 
+# %%
+# Plot Axx
 mesh.plot(aXX, title="Axx", text="Axx plot")
-mesh.plot(aYY, title="Ayy", text="Ayy plot")
-mesh.plot(aXZ, title="Axz", text="Axz plot")
+
+
+# %%
+# Plot Ayy
+mesh2.plot(aYY, title="Ayy", text="Ayy plot")
+
+
+# %%
+# Plot Axz
+mesh3.plot(aXZ, title="Axz", text="Axz plot")
```

**Comment by PProfizi** [comment]
Hi @greschd @FedericoNegri, I did not manage to reproduce the issue, even with the three different meshes. I am using:
- Windows
- Python 3.10
- This branch of pydpf-composites
- I tried with Ansys 231 (DPF 5.0) and the latest 232 (DPF 6.1)
- I have this list of packages installed:
```
ansys-dpf-composites     0.2.dev0  D:\ANSYSDev\Sandbox\pydpf-composites
ansys-dpf-core           0.7.4                                         
ansys-dpf-gate           0.3.1                                         
ansys-dpf-gatebin        0.3.1                                         
ansys-grpc-dpf           0.7.1                                         
appdirs                  1.4.4                                         
argcomplete              3.0.5
cachetools               5.3.0
certifi                  2022.12.7
charset-normalizer       3.1.0
click                    8.1.3
colorama                 0.4.6
contourpy                1.0.7
cycler                   0.11.0
fonttools                4.39.3
google-api-core          2.11.0
google-api-python-client 2.83.0
google-auth              2.17.1
google-auth-httplib2     0.1.0
googleapis-common-protos 1.59.0
grpcio                   1.53.0
httplib2                 0.22.0
idna                     3.4
imageio                  2.27.0
importlib-metadata       6.1.0
kiwisolver               1.4.4
matplotlib               3.7.1
numpy                    1.24.2
packaging                23.0
Pillow                   9.5.0
pip                      21.3.1
pipx                     1.2.0
protobuf                 4.22.1
psutil                   5.9.4
pyasn1                   0.4.8
pyasn1-modules           0.2.8
pyparsing                3.0.9
python-dateutil          2.8.2
pyvista                  0.36.1
requests                 2.28.2
rsa                      4.9
scipy                    1.10.1
scooby                   0.7.1
setuptools               60.2.0
six                      1.16.0
tqdm                     4.65.0
uritemplate              4.1.1
urllib3                  1.26.15
userpath                 1.8.0
vtk                      9.2.6
wheel                    0.37.1
zipp                     3.15.0
```
Is there something I am missing?

**Comment by greschd** [comment]
@PProfizi I realized the difference is not due to the three different meshes, but due to running via the `pydpf-composites` Docker container or directly from the unified install.

The unified install gives the correct output, whereas the Docker container does not. I'm not sure if this means it's a Linux-only issue, or if something is wrong only in the Docker container. I'll test a unified install on Linux to compare.

**Comment by greschd** [comment]
The difference seems to be independent of plotting. I've modified the example to just print a result, and run with both a local and container server [here](https://github.com/pyansys/pydpf-composites/blob/dbg/fiber_orientation_example/examples/9_short_fiber_orientation_tensor.py)

Output on Windows:
```
<local server output>
Result local:  0.8476316975728202
Result container:  0.08833158599895692
```

File: examples/9_short_fiber_orientation_tensor.py (hunk 3 above)
Comment (FedericoNegri) [reply examples/9_short_fiber_orientation_tensor.py:28]: Agree, thanks. Added a note.

**Comment by greschd** [comment]
@PProfizi FYI, we figured out the root cause for these issues. In the example, we're accessing some data by index instead of by label, and the order in which the arrays are added is not deterministic.

No further action needed from your side, thanks for your help.

**Comment by greschd** [comment]
@FedericoNegri please review the changes made to the example. The main change is using the field variable name to correctly identify them (since the index is non-deterministic).

**Pull request merged**